### PR TITLE
[feat]: [CI-10145]: Contact between ML CLI and TI Service

### PR DESCRIPTION
**Pull request by anurag-harness**

### CODE REVIEW

File: client/http.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ const (
 	getTestsTimesEndpoint = "/tests/timedata?accountId=%s&orgId=%s&projectId=%s&pipelineId=%s"
 	agentEndpoint         = "/agents/link?accountId=%s&language=%s&os=%s&arch=%s&framework=%s&version=%s&buildenv=%s"
 	commitInfoEndpoint    = "/vcs/commitinfo?accountId=%s&orgId=%s&projectId=%s&pipelineId=%s&buildId=%s&stageId=%s&stepId=%s&repo=%s&branch=%s"
+	mlSelectTestsEndpoint = "/ml/select_tests?accountId=%s&orgId=%s&projectId=%s&pipelineId=%s&buildId=%s&stageId=%s&stepId=%s&repo=%s&mlKey=%s&target=%s"
```
Comment (rutvijmehta-harness) [inline client/http.go:36]: Check if you require all of these fields in the API

File: types/types.go [hunk 2]
```diff
@@ -302,3 +302,28 @@ type GetCgCountResp struct {
...
 	LastSuccessfulCommitId string `json:"commit_id"`
 }
+
+// ML Based Test Selection Request and Response
+type MLSelectTestsRequest struct {
+	MLServiceAPIRequest MLServiceAPIRequest `json:"ml_service_api_request"`
+	Percentile          int                 `json:"percentile"`
+}
+
+type MLServiceAPIRequest struct {
```
Comment (rutvijmehta-harness) [inline types/types.go:312]: This should go to types.go of ML client.

Comment (anurag-harness) [reply types/types.go:312]: Do you know how I can import it here using an scm url then?

Comment (anurag-harness) [reply types/types.go:312]: I'll keep it here and in ml-client code for now. Once ti-service changes are merged to github i'll remove it from here and use the struct in ml-client in the ml-ti-cli. Is that okay?

Comment (rutvijmehta-harness) [reply types/types.go:312]: Let's discuss this tomorrow.

File: client/http.go (hunk 1 above)
Comment (anurag-harness) [reply client/http.go:36]: Dont need all but I think we should keep them since it helps in debugging

File: types/types.go (hunk 2 above)
Comment (anurag-harness) [reply types/types.go:312]: As discussed, keeping this here as well as in ML Client with a manual copy to ML Client

**Review by jamie-harness — APPROVED** [review]

**Pull request merged**